### PR TITLE
[ENG-3609] fix: ordinal image should fall back to the ordiview thumbnail

### PR DESCRIPTION
**Pull request by teebszet**
# 🔘 PR Type
- [x] Bugfix
- [x] Enhancement

# 📜 Background
Issue Link: [ENG-3609](https://linear.app/xverseapp/issue/ENG-3609/change-how-we-show-unknown-inscriptions-in-the-apps)

# 🔄 Changes
changes:
- ordinal image should fall back to the ordiview thumbnail for empty/undefined content types
- style application/json content types for inscription assets
- add the brc20 validation check before rendering brc20 thumbnails
- add lim and dec props to brc20 deploy inscriptions

impact:
- ordinal image is used in a lot of places
```
src/app/screens/restoreFunds/restoreOrdinals/ordinalRow.tsx:81:        <OrdinalImage isSmallImage withoutSizeIncrease ordinal={ordinalData?.data!} />
src/app/screens/ordinals/index.tsx:60:          <OrdinalImage isNftDashboard ordinal={asset} />
src/app/screens/ordinals/index.tsx:63:        <OrdinalImage isNftDashboard ordinal={asset} />
src/app/screens/nftDashboard/inscriptionsTabGridItem.tsx:75:          <OrdinalImage ordinal={{ id: '', content_type: 'unknown', number: 0 }} />
src/app/screens/nftDashboard/inscriptionsTabGridItem.tsx:77:          <OrdinalImage ordinal={collection.thumbnail_inscriptions[0]} />
src/app/screens/nftDashboard/inscriptionsTabGridItem.tsx:79:          <OrdinalImage ordinal={collection.thumbnail_inscriptions[0]} withoutTitles />
src/app/screens/ordinalsCollection/index.tsx:246:                    <OrdinalImage ordinal={inscription} />
src/app/screens/confirmOrdinalTransaction/index.tsx:178:              <OrdinalImage inNftSend withoutSizeIncrease ordinal={selectedOrdinal!} />
src/app/screens/ordinalDetail/index.tsx:670:        <OrdinalImage ordinal={ordinal!} />
src/app/screens/ordinalDetail/index.tsx:753:            <OrdinalImage ordinal={ordinal!} inNftDetail />
src/app/components/collectibleCollage/collectibleCollage.tsx:59:            <OrdinalImage ordinal={item as CondensedInscription} />
src/app/components/assetModal/index.tsx:81:          <OrdinalImage ordinal={consdensedInscription} />
src/app/components/confirmBtcTransaction/itemRow/inscription.tsx:60:            <OrdinalImage
```

# 🖼 Screenshot / 📹 Video
before:
image

after:
image

image

https://github.com/secretkeylabs/xverse-web-extension/assets/6109710/db1eb646-b6dd-49b0-b9d4-d37a2f549c9a

# ✅ Review checklist
Please ensure the following are true before merging:

- [ ] Code Style is consistent with the project guidelines.
- [ ] Code is readable and well-commented.
- [ ] No unnecessary or debugging code has been added.
- [ ] Security considerations have been taken into account.
- [ ] The change has been manually tested and works as expected.
- [ ] Breaking changes and their impacts have been considered and documented.
- [ ] Code does not introduce new technical debt or issues.

### CODE REVIEW

**Review by victorkirov — COMMENTED** [review]
Any chance we can define a display for the application json ones? e.g. 606466 from the screenshot. That should actually show a brc-20 icon 🤔

**Comment by teebszet** [comment]
> Any chance we can define a display for the application json ones? e.g. 606466 from the screenshot. That should actually show a brc-20 icon 🤔

I believe it's because this one doesn't pass our checks for a valid brc20: https://ord-testnet.xverse.app/inscription/0dfd1d8cc0e4f558ad5b36823b2cb2b5455733a20dfed707b4f47bef332b6515i0

we check tick is 4 chars only

File: src/app/screens/ordinals/ordinalImage.tsx [hunk 1]
```diff
@@ -237,7 +237,8 @@ function OrdinalImage({
     );
   }
 
-  if (contentType.includes('image')) {
+  // if content type is undefined or "", we treat it the same as an image and fall back to ordiview thumbnail
+  if (!contentType || contentType.includes('image')) {
     return renderImage(t('ORDINAL'), `${XVERSE_ORDIVIEW_URL(network.type)}/content/${ordinal.id}`);
   }
```
Comment (victorkirov) [inline src/app/screens/ordinals/ordinalImage.tsx:243]:
```suggestion
  if (contentType.includes('image')) {
    return renderImage(t('ORDINAL'), `${XVERSE_ORDIVIEW_URL(network.type)}/content/${ordinal.id}`);
  }
  
  // if content type is undefined or "", we fall back to ordiview thumbnail
  if (!contentType) {
    return renderImage(t('ORDINAL'), `${XVERSE_ORDIVIEW_URL(network.type)}/thumbnail/${ordinal.id}`);
  }
```

Comment (teebszet) [reply src/app/screens/ordinals/ordinalImage.tsx]: thanks for catching that

**Review by victorkirov — DISMISSED** [review]

**Review by victorkirov — APPROVED** [review]

**Review by victorkirov — APPROVED** [review]

**Comment by DuskaT021** [comment]
this can be merged

**Pull request merged**